File: Baekjoon/Level8/8-02/main.c
#include <stdio.h>

int main() {
    long long n;
    scanf("%lld", &n);
    int count = 1;
    long long range = 1;
    long long tmp = 1;
    while (1) {
        if (range >= n) break;
        tmp = 6 * (count++);
        range += tmp;
    }
    printf("%d", count);
    return 0;
}

File: Baekjoon/Level8/8-02/main.dart
import 'dart:io';

int solution(int n) {
  int answer = 1;
  int range = 1, tmp = 1;
  while (true) {
    if (range >= n) break;
    tmp = 6 * (answer++);
    range += tmp;
  }
  return answer;
}

void main() {
  int n = int.parse(stdin.readLineSync()!);

  print(solution(n));
}
